packages/examples/cvat/recording-oracle/src/crons/process_exchange_oracle_webhooks.py

```diff
@@ -1,135 +1,61 @@
 
 
-def process_incoming_exchange_oracle_webhooks():
-    """
-    Process incoming oracle webhooks
-    """
-    logger = get_function_logger(module_logger_name)
+@cron_job(module_logger_name)
+def process_incoming_exchange_oracle_webhooks(logger: logging.Logger, session: Session):
+    webhooks = oracle_db_service.inbox.get_pending_webhooks(
+        session,
+        OracleWebhookTypes.exchange_oracle,
+        limit=Config.cron_config.process_exchange_oracle_webhooks_chunk_size,
+        for_update=ForUpdateParams(skip_locked=True),
+    )
 
-    try:
-        logger.debug("Starting cron job")
+    for webhook in webhooks:
+        with handle_webhook(logger, session, webhook):
+            handle_exchange_oracle_event(webhook, db_session=session)
 
-        with SessionLocal.begin() as session:
-            webhooks = oracle_db_service.inbox.get_pending_webhooks(
-                session,
-                OracleWebhookTypes.exchange_oracle,
-                limit=Config.cron_config.process_exchange_oracle_webhooks_chunk_size,
-                for_update=ForUpdateParams(skip_locked=True),
-            )
-
-            for webhook in webhooks:
-                try:
-                    logger.debug(
-                        "Processing webhook "
-                        f"{webhook.type}.{webhook.event_type}~{webhook.signature} "
-                        f"in escrow_address={webhook.escrow_address} "
-                        f"(attempt {webhook.attempts + 1})"
-                    )
-
-                    handle_exchange_oracle_event(webhook, db_session=session, logger=logger)
-
-                    oracle_db_service.inbox.handle_webhook_success(session, webhook.id)
-                    logger.debug("Webhook handled successfully")
-                except Exception as e:
-                    logger.exception(f"Webhook {webhook.id} handling failed: {e}")
-                    oracle_db_service.inbox.handle_webhook_fail(session, webhook.id)
-    except Exception as e:
-        logger.exception(e)
-    finally:
-        logger.debug("Finishing cron job")
 
-
-def handle_exchange_oracle_event(webhook: Webhook, *, db_session: Session, logger: logging.Logger):
+def handle_exchange_oracle_event(webhook: Webhook, *, db_session: Session):
     assert webhook.type == OracleWebhookTypes.exchange_oracle
 
     match webhook.event_type:
         case ExchangeOracleEventTypes.task_finished:
-            logger.debug(
-                f"Received a task finish event for escrow_address={webhook.escrow_address}. "
-                "Validating the results"
-            )
-
             validate_results(
                 escrow_address=webhook.escrow_address,
                 chain_id=webhook.chain_id,
                 db_session=db_session,
             )
-
+        case ExchangeOracleEventTypes.escrow_cleaned:
+            clean_escrow(escrow_address=webhook.escrow_address, chain_id=webhook.chain_id)
+            mark_escrow_as_cleared(
```

Comment:
1. Probably, it should be a single function
2. Please be consistent about using clean and clear